onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -395,9 +395,24 @@ Status Predict(const std::unordered_map<std::string, OnnxTensorData>& inputs,
       compiled_model_path_ = [compileUrl path];
 
       MLModelConfiguration* config = [[MLModelConfiguration alloc] init];
-      config.computeUnits = (coreml_flags_ & COREML_FLAG_USE_CPU_ONLY)
-                                ? MLComputeUnitsCPUOnly
-                                : MLComputeUnitsAll;
+      uint32_t device_flags = coreml_flags_ & (COREML_FLAG_USE_CPU_ONLY | COREML_FLAG_USE_CPU_AND_GPU);
+      // check if only one flag is set
+      if (device_flags & (device_flags - 1)) {
+        // multiple device options selected
+        return ORT_MAKE_STATUS(
+            ONNXRUNTIME, FAIL,
+            "Multiple device options selected, "
+            "you should use at most one of the following options: COREML_FLAG_USE_CPU_ONLY or COREML_FLAG_USE_CPU_AND_GPU");
+      }
```

Comment:
Can we do this check in the C++ code when creating the EP as that is the point where the user options are being passed in? 

It's also easier to maintain/test as that code can run on any platform.

Reply:
Sounds good

Reply:
Moved to the EP construct function.